src/main/java/com/google/flourbot/entity/EntityModuleImplementation.java

```diff
@@ -106,7 +106,7 @@ public EntityModuleImplementation(DataStorage dataStorage) {
             if (row == null) {
               throw new IllegalStateException("row not found in Firestore action");
             }
-            action = new SheetReadRowAction(sheetUrl, (int) row);
+            action = new SheetReadRowAction(sheetUrl, (int) row, (boolean) actionData.get("selectRandomEntry"));
```

Comment:
This works for now, but it doesn't scale, I think it would be better to have a SelectionMethod enum that contains other options, like first, last etc

Reply:
Also, please see my comment below, but I think this approach is not intuitive. The action is called ReadRow, but depending on whether selectRandomEntry is false/true you will either return the full row or just one field.

I think it would be better to have the enum as described above and an int variable called limit. -1 would mean no limit. This way you can define results like 
FIRST -1 (all fields in the natural order - the default)
LAST 3 - the last 3 fields reverted from the end
RANDOM 1 - single random field (which is what you have here)
This way you preserve the nature of the action (i.e. get the list of fields in the row) but also introduce some flexibility.
For now you can just hardcode two options: FIRST -1 and RANDOM 1 and pick them depending on the state of selectRandomEntry - later we can fix the UI.

Reply:
I love the enum idea! Added it. Will need to code the options later - if we judge it is important
Note that I moved the handling logic to DriveCloudSheet since I thought it was appropriate to have there, but if you think it's better to have within the SheetReadRowAction and SheetReadColumnAction let me know